libraries/chain/include/graphene/chain/database.hpp

```diff
@@ -424,12 +445,29 @@ namespace graphene { namespace chain {
                    const price& fill_price);
 
          /**
+          * @brief fills limit order
+          * @param order the order
+          * @param pays what the account is paying
+          * @param receives what the account is receiving
+          * @param cull_if_small take care of dust
+          * @param fill_price the transaction price
+          * @param is_maker TRUE if this order is maker, FALSE if taker
           * @return true if the order was completely filled and thus freed.
           */
-         bool fill_limit_order( const limit_order_object& order, const asset& pays, const asset& receives, bool cull_if_small,
-                                const price& fill_price, const bool is_maker );
+         bool fill_limit_order( const limit_order_object& order, const asset& pays, const asset& receives,
+               bool cull_if_small, const price& fill_price, const bool is_maker );
+         /***
+          * @brief attempt to fill a call order
+          * @param order the order
+          * @param pays what the buyer pays for the collateral
+          * @param receives the collateral received by the buyer
+          * @param fill_price the price the transaction executed at
+          * @param is_maker TRUE if the buyer is the maker, FALSE if the buyer is the taker
+          * @param margin_fee Margin call fees paid in collateral asset
+          * @returns TRUE if the order was completely filled
+          */
          bool fill_call_order( const call_order_object& order, const asset& pays, const asset& receives,
-                               const price& fill_price, const bool is_maker );
+                               const price& fill_price, const bool is_maker, const asset& margin_fee = asset(0) );
```

Comment:
The default value changes old data.

The old code stores `asset(0,pays.asset_id)` to the `fee` field of the virtual `fill_order_operation`, but the new code will store `asset(0,asset_id_type(0))` which could have a different asset ID.

https://github.com/bitshares/bitshares-core/blob/ed6b6d8dc93d266407959d342235aaa740d64dc5/libraries/chain/db_market.cpp#L923-L924

https://github.com/bitshares/bitshares-core/blob/11aa1bea71d4e62404e04b2dc7c7600847d004b8/libraries/chain/db_market.cpp#L1003-L1004